### DISCUSSION TITLE
make the function `print` return width and height

### DISCUSSION THREAD

**Discussion by bettywolfy** [nesbox/TIC-80#2496] [Ideas]
why not?
It's difficult to calculate the right size, especially when `\n` is in the text.

**Comment by Skeptim** [comment]
Hi,
That's already the case ^^ partly. It returns the text width https://github.com/nesbox/TIC-80/wiki/print
The text height is scale*6 pixels, not sure it's useful to add it as output but I added it to the wiki.

**Reply by bettywolfy** [reply]
`"Hello!\nWorld!"`
I would need to know how many `\n` are in the string.

**Reply by Skeptim** [reply]
Ah, indeed ! Sorry, I never use \n because there is no interline.
In lua you can do something like
```lua
n_count=0
text="HELLO\nWOR\nLD!"	
for char_id=1,#text,1 do
  if text:sub(char_id,char_id)=="\n" then
    n_count=n_count+1
  end
end
trace(n_count)
```

Or you can copy-paste the [extended print function](https://github.com/nesbox/TIC-80/wiki/print#extended-print-function) that outputs the text height (and has interline and many other useful stuff).

**Reply by bettywolfy** [reply]
the code appears to be slow.

**Reply by Skeptim** [reply]
That's a lua related question, I haven't tested but this seems better: https://stackoverflow.com/questions/11152220/counting-number-of-string-occurrences

**Reply by joshgoebel** [reply]
> the code appears to be slow.

How slow?
